src/java.base/share/classes/java/security/spec/PSSParameterSpec.java

```diff
@@ -102,9 +107,15 @@
 
     /**
      * The PSS parameter set with all default values
+     * @deprecated The default values defined in the ASN.1 encoding in
+     *         PKCS#1 may become obsolete as time progresses. Thus, it is
+     *         recommended to explicitly specify all desired parameter
+     *         values with
+     *         {@link #PSSParameterSpec(String, String, AlgorithmParameterSpec, int, int) PSSParameterSpec}.
```

Comment:
We are deprecating a field so I would say "This field uses default values defined in ... which may become...".

Do we need to write "PKCS #1" with a blank inside? Same below.

Also, the "all desired parameter values" phrase is perfect for the constructor below but this is for a field not a method so "parameters" does not make sense to me. How about something like "Instead of using this field, user should create a new `PSSParameterSpec` object by calling..." or we can just not mention it. User would need to create one anyway.

Reply:
Well, I did a quick search, it seems both PKCS#X and PKCS #X are used interchangeably. I use PKCS#X to be consistent within this file. Maybe it does not matter much? I thought we tend to return without the space, e.g. PKCS8EncodedKeySpec.getFormat(). Other wording suggestions incorporated. Thanks~

Reply:
My understanding is that if there is a "#" then there should be a " " before it. However, if the format without the space is already used elsewhere, we can keep using it.